### PR TITLE
Add smart_proxy_remote_execution_ssh (RPM)

### PR DESCRIPTION
**Pull request by stbenjam**

### CODE REVIEW

**Comment by stbenjam** [comment]
Checklist:

- [ ] algebrick https://github.com/theforeman/foreman-packaging/pull/769
- [ ] concurrent-ruby
- [ ] concurrent-ruby-edge	
- [ ] apipie-params
- [ ] dynflow
- [ ] smart_proxy_dynflow https://github.com/theforeman/foreman-packaging/pull/740
- [ ] smart_proxy_remote_execution_ssh. This PR!

**Comment by stbenjam** [comment]
Since I already started with separate PR's, I'll leave it that way unless you want it changed... I'll remember I can do multiple packages per PR next time

**Comment by stbenjam** [comment]
Need to remove the Gemfile from the spec, and add the README. 

https://github.com/theforeman/smart_proxy_remote_execution_ssh/pull/3

**Comment by stbenjam** [comment]
Everyone's green

File: rubygem-smart_proxy_remote_execution_ssh/rubygem-smart_proxy_remote_execution_ssh.spec [hunk 1]
```diff
@@ -0,0 +1,87 @@
+%global gem_name smart_proxy_remote_execution_ssh
+
+%global foreman_proxy_dir /usr/share/foreman-proxy
+%global foreman_proxy_bundlerd_dir %{foreman_proxy_dir}/bundler.d
+%global foreman_proxy_settingsd_dir %{_sysconfdir}/foreman-proxy/settings.d
+
+%global remote_execution_ssh_config_dir %{_sysconfdir}/remote_execution_ssh
```
Comment (domcleal) [inline rubygem-smart_proxy_remote_execution_ssh/rubygem-smart_proxy_remote_execution_ssh.spec]: What's this for?

File: rubygem-smart_proxy_remote_execution_ssh/rubygem-smart_proxy_remote_execution_ssh.spec [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+Name: rubygem-%{gem_name}
+Version: 0.0.2
+Release: 1%{?dist}
+Group: Applications/System
+License: GPLv3
+URL: https://github.com/theforeman/smart_proxy_remote_execution_ssh
+Source0: http://rubygems.org/downloads/%{gem_name}-%{version}.gem
+
+Requires: foreman-proxy >= 1.9.0
+Requires: rubygem(smart_proxy_dynflow)
```
Comment (domcleal) [inline rubygem-smart_proxy_remote_execution_ssh/rubygem-smart_proxy_remote_execution_ssh.spec]: ~> 0.0.3

File: rubygem-smart_proxy_remote_execution_ssh/rubygem-smart_proxy_remote_execution_ssh.spec [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+BuildRequires: ruby(abi)
+%else
+Requires: ruby(release)
+BuildRequires: ruby(release)
+%endif
+BuildRequires: rubygems-devel
+BuildArch: noarch
+
+Provides: rubygem(%{gem_name}) = %{version}
+Provides: foreman-proxy-plugin-remote_execution_ssh
```
Comment (domcleal) [inline rubygem-smart_proxy_remote_execution_ssh/rubygem-smart_proxy_remote_execution_ssh.spec]: I'll admit we don't really use this, but usually we hyphenate it all

File: rubygem-smart_proxy_remote_execution_ssh/rubygem-smart_proxy_remote_execution_ssh.spec [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+%install
+mkdir -p %{buildroot}%{gem_dir}
+
+cp -pa .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
+
+mkdir -p %{buildroot}%{foreman_proxy_bundlerd_dir}
+cp -pa .%{gem_instdir}/bundler.d/remote_execution_ssh.rb %{buildroot}%{foreman_proxy_bundlerd_dir}
+mkdir -p  %{buildroot}%{foreman_proxy_settingsd_dir}
+cp -pa .%{gem_instdir}/settings.d/ssh.yml.example %{buildroot}%{foreman_proxy_settingsd_dir}/ssh.yml
```
Comment (domcleal) [inline rubygem-smart_proxy_remote_execution_ssh/rubygem-smart_proxy_remote_execution_ssh.spec]: This should be called remote_execution_ssh I think, it'll certainly help to be consistent if it's added to the foreman_proxy Puppet module.

File: comps/comps-foreman-plugins-rhel7.xml [hunk 5]
```diff
@@ -53,6 +53,7 @@
       <packagereq type="default">rubygem-smart_proxy_chef</packagereq>
       <packagereq type="default">rubygem-smart_proxy_pulp</packagereq>
       <packagereq type="default">rubygem-smart_proxy_salt</packagereq>
+      <packagereq type="default">rubygem-smart_proxy_remote_execution_ssh</packagereq>
```
Comment (domcleal) [inline comps/comps-foreman-plugins-rhel7.xml]: remote comes before salt

File: rubygem-smart_proxy_remote_execution_ssh/rubygem-smart_proxy_remote_execution_ssh.spec (hunk 1 above)
Comment (stbenjam) [inline rubygem-smart_proxy_remote_execution_ssh/rubygem-smart_proxy_remote_execution_ssh.spec]: Just seeing if you're paying attention :grin:

File: rubygem-smart_proxy_remote_execution_ssh/rubygem-smart_proxy_remote_execution_ssh.spec (hunk 4 above)
Comment (stbenjam) [inline rubygem-smart_proxy_remote_execution_ssh/rubygem-smart_proxy_remote_execution_ssh.spec]: Indeed, added to https://github.com/theforeman/smart_proxy_remote_execution_ssh/pull/3

Will need to update the gem.

**Comment by stbenjam** [comment]
Green, updated to 0.0.3 with right yml

**Comment by domcleal** [comment]
Merged and building, thanks @stbenjam.

**Pull request merged**